File: LeetCode-contest-234/5731.cc
class Solution {
public:
  int numDifferentIntegers(string word) {
    unordered_set<long> s;
    word += 'a';
    long num = 0;
    char prev = 'a';
    for (char c : word) {
      if (isdigit(c)) {
        num = 10 * num + (c - '0');
      } else {
        if (num == 0 && prev == '0')
          s.insert(0);
        else if (num != 0)
          s.insert(num);
        num = 0;
      }
      prev = c;
    }
    return s.size();
  }
};

File: LeetCode-Contests/LeetCode-contest-234/5731.py
class Solution:
    def numDifferentIntegers(self, word: str) -> int:
        word += 'a'
        s = set()
        num = 0
        prev = ''
        for c in word:
            if c.isdigit():
                num = 10 * num + int(c)
            else:
                if num == 0 and prev == '0':
                    s.add(0)
                elif num != 0:
                    s.add(num)
                num = 0
            prev = c
        return len(s)
